### PR TITLE
1266480: Refresh TreeView selection after subscriptions are removed.

### PR DESCRIPTION
**Pull request by awood**
I'm a little *meh* about this fix, but given time constraints, I think it will have to suffice.  I don't like that when you unsubscribe an item say 4 rows down in the TreeView that the selection jumps back up to row 0 after the unsubscribe is finished.

I tried

```python
selection.store.remove(selection.tree_iter)
```

which moves the selection up one row, but something else is running shortly thereafter that moves the selection back to 0.

### CODE REVIEW

**Comment by alikins** [comment]
tl;dr Looks ok to me, considering.

Looks like a lot of things could be triggering rebuilding the model (and widget and selection).

MySubscriptionsTab._unsubscribe_callback gets added to idle loop by async_bind.unbind, so that
will happen at some point later. and _ubsubscribe_callback calls self.backend.cs.force_cert_check()
directly, but the timer used for CertSorter's cert_monitor will also happen at a regular interval
(and if the cert_monitor triggers any callbacks added to CertSorter)

I think resetting the selection is reasonable, given expectations are a little undefined.

Maybe connect to the View's select 'changed' signal and track the current select (and position in the order?), and attempt to reset the selection to something close after update_subscriptions?
That likely means:
```
        # 841396: Select first item in My Subscriptions table by default
        selection = self.top_view.get_selection()
        selection.select_path(0)
```
would need to move out of update_subscriptions

**Comment by alikins** [comment]
re selection 'changed':
ala http://python-gtk-3-tutorial.readthedocs.org/en/latest/treeview.html#the-selection

**Comment by awood** [comment]
Yeah, I think getting it right every time would require a lot of effort.  I'd rather the selection just do something non-optimal but defined ("go to the top") than work most of the time but occasionally do something bizarre.

**Comment by alikins** [comment]
After reproducing this, it looks like the tree views selection 'changed' signal already has a handler (MySubsTab.on_selection). And thats getting triggered with a now bogus selection (a serial that can't be found) which likely means the 'changed' is emitted twice (or more). Probably when the selection gets reset after the store.clear in update_subscriptions? 

I'd suspect that 'changed' handler needs to be blocked when updating subscriptions, but that means changes in widgets.SubscriptionManagerTab and friends. Something for later.

**Comment by alikins** [comment]
ACK
